### PR TITLE
Fix default discovery crash on windows

### PR DESCRIPTION
**Pull request by pjanotti**
**Description:**
A bad default on the OTel `docker_host` (to be fixed upstream) exposed that if a observer extension fails to `Start` the discovery code still calls `ListAndWatch` what causes a panic.

Details: the Start of the observer is called via internal/confmapprovider/discovery, but if Start fails it logs a warning and proceeds with the execution. Later internal/receiver/discoveryreceiver uses the host.GetExtensions()  and the observer is there since it was created successfully (it did fail to start in a step that the host doesn't have visibility). The extension is cast to an observer.Observable and a go routine calling ListAndWatch is launched, but, then the private that it needs to run are nil because the start failed and then panic.

Right now I am removing the `docker_observer` from Windows, but, it would be re-enabled as soon as we have it fixed on upstream.

**Jira:** 
https://splunk.atlassian.net/browse/OTL-2883

**Testing:**
Added 2 new tests and validated manually.

**Documentation:**
N/A

### CODE REVIEW

**Comment by pjanotti** [comment]
Perhaps that some intermittent error that we got on Windows runners when some ports that we use are already taken? Not sure at this point... kicking it again: https://github.com/signalfx/splunk-otel-collector/actions/runs/10184803130/job/28172948607?pr=5162#step:5:386

**Comment by pjanotti** [comment]
The port failure is not a Windows only thing it also affects mac, fixing via https://github.com/signalfx/splunk-otel-collector/pull/5168

**Comment by pjanotti** [comment]
https://github.com/signalfx/splunk-otel-collector/pull/5168 was not enough... also failed on Ubuntu, it may need a refactor of the actual type.

File: cmd/otelcol/main.go [hunk 1]
```diff
@@ -80,14 +85,21 @@ func main() {
 		},
 	}
 
-	os.Args = append(os.Args[:1], collectorSettings.ColCoreArgs()...)
+	allArgs := args[:1]
+	allArgs = append(allArgs, collectorSettings.ColCoreArgs()...)
+	os.Args = allArgs
```
Comment (dmitryax) [inline cmd/otelcol/main.go:90]: why this change? can't see a difference

File: cmd/otelcol/main.go [hunk 2]
```diff
@@ -36,10 +37,14 @@ import (
 )
 
 func main() {
+	runFromCmdLine(os.Args)
+}
+
+func runFromCmdLine(args []string) {
```
[the remark this answers is not in the document]
Comment (pjanotti) [reply cmd/otelcol/main.go:43]: Testability, this way I can call `runFromCmdLine` directly from tests.

File: cmd/otelcol/main.go (hunk 1 above)
Comment (pjanotti) [reply cmd/otelcol/main.go:90]: Lint doesn't like `append` that doesn't assign resulting slice to the first parameter so logically what I want is `os.Args = append(args[:1], collectorSettings.ColCoreArgs()...)`, but, lint is not happy that `args` is not on the left-hand side.

**Comment by pjanotti** [comment]
Working to fix the test failure via https://github.com/signalfx/splunk-otel-collector/pull/5189

**Comment by pjanotti** [comment]
Still having issues with the new test affecting other tests on the same test executable, switching to draft until that is resolved.

**Comment by pjanotti** [comment]
I will create a new PR to do the actual fix after https://github.com/signalfx/splunk-otel-collector/pull/5201 is merged. This will reduce the review scope and will make the history better.
